### PR TITLE
Enhanced RDoc for FileUtils

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Treats:
- ::pwd
- ::cd
- ::uptodate?
- ::mkdir

### CODE REVIEW

File: lib/fileutils.rb [hunk 1]
```diff
@@ -146,11 +168,14 @@ def cd(dir, verbose: nil, &block) # :yield: dir
   module_function :chdir
 
   #
-  # Returns true if +new+ is newer than all +old_list+.
-  # Non-existent files are older than any file.
+  # Returns +true+ if the file at path +new+
+  # is newer than all the files at paths in array +old_list+.;
```
Comment (peterzhu2118) [inline lib/fileutils.rb:172]: I think the period is a typo

```suggestion
  # is newer than all the files at paths in array +old_list+;
```

File: lib/fileutils.rb [hunk 2]
```diff
@@ -170,12 +195,32 @@ def remove_trailing_slash(dir)   #:nodoc:
   private_module_function :remove_trailing_slash
 
   #
-  # Creates one or more directories.
+  # Creates directories at the paths in the given +list+ (an array of strings);
+  # returns +list+.
```
Comment (peterzhu2118) [inline lib/fileutils.rb:199]: I think we should mention that `list` could also be a string (to create just a single directory).

File: lib/fileutils.rb (hunk 1 above)
Comment (BurdetteLamar) [reply lib/fileutils.rb]: Fixed.

File: lib/fileutils.rb (hunk 2 above)
Comment (BurdetteLamar) [reply lib/fileutils.rb]: Done.

**Review by peterzhu2118 — APPROVED** [review]
Looks good!

**Pull request merged**